types/dynamic_fee_tx.go

```diff
@@ -33,6 +33,21 @@ func (tx *DynamicFeeTx) gasPrice() *big.Int      { return nil }
 func (tx *DynamicFeeTx) gasTipCap() *big.Int     { return tx.GasTipCap }
 func (tx *DynamicFeeTx) gasFeeCap() *big.Int     { return tx.GasFeeCap }
 
+func (tx *DynamicFeeTx) effectiveGasPrice(baseFee *big.Int) *big.Int {
+	var tmp = new(big.Int)
+
+	if baseFee == nil {
+		return tmp.Set(tx.GasFeeCap)
+	}
+
+	tip := tmp.Sub(tx.GasFeeCap, baseFee)
+	if tip.Cmp(tx.GasTipCap) > 0 {
+		tip.Set(tx.GasTipCap)
+	}
+
+	return tip.Add(tip, baseFee)
```

Comment:
```suggestion
	tmp.Sub(tx.GasFeeCap, baseFee)
	if tmp.Cmp(tx.GasTipCap) > 0 {
		tmp.Set(tx.GasTipCap)
	}

	return tmp.Add(tmp, baseFee)
```
IMO this can be a little bit simplified in order to avoid tip variable.